@navikt/core/react/src/layout/base/PrimitiveWithAsChild.ts

```diff
@@ -0,0 +1,40 @@
+export type PrimitiveWithAsChild =
+  | {
+      children: React.ReactElement | false | null;
+      /**
+       * Renders the component and its child as a single element,
+       * merging the props of the component with the props of the child.
+       * @example
+       * ```tsx
+       * <Component asChild data-prop>
+       *   <ChildComponent data-child />
+       * </Component>
+       *
+       * // Renders
+       * <div data-prop data-child />
+       * ```
+       */
+      asChild: true;
+      /**
+       * When using asChild, the `as` prop is not allowed.
+       */
+      as?: never;
```

Comment:
Getting this now. Since "as" is left out in the first type-def its set to type: undefined 🤔 
```json
"as": {
        "defaultValue": null,
        "description": "When using asChild, the `as` prop is not allowed.",
        "name": "as",
        "declarations": [
          {
            "fileName": "react/src/layout/base/PrimitiveAsChildProps.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "undefined"
        }
      }
```

By changing to `as?: React.ElementType;` this was the output 
```
"as": {
        "defaultValue": null,
        "description": "When using asChild, the `as` prop is not allowed.",
        "name": "as",
        "declarations": [
          {
            "fileName": "react/src/layout/base/PrimitiveAsChildProps.ts",
            "name": "TypeLiteral"
          },
          {
            "fileName": "react/src/layout/base/PrimitiveAsChildProps.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "ElementType<any, keyof IntrinsicElements>"
        }
      }
```
buuut... That broke TS since the type got to complex to handle `Expression produces a union type that is too complex to represent.`

Suggested fix: add filter for undefined "as"-types in docgen.ts
```
if (prop.name === "as" && prop.type.name === "undefined") {
    return false;
}
```
Since we can manually add the `as` documentation with a switch in sanity, this will not remove needed documentation.

Reply:
Do it